src/overlays/actors/ovl_Obj_Chikuwa/z_obj_chikuwa.c

```diff
@@ -28,31 +28,206 @@ const ActorInit Obj_Chikuwa_InitVars = {
+    } else {
+        func_800C62BC(play, &play->colCtx.dyna, this->dyna.bgId);
+    }
+}
+
+void ObjChikuwa_Init(Actor* thisx, PlayState* play) {
+    ObjChikuwa* this = THIS;
+    s32 i;
+    s32 val;
+    ObjChikuwaStruct* temp;
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+
+    this->dyna.actor.scale.x = 0.15f;
+    this->dyna.actor.scale.y = 0.2f;
+    this->dyna.actor.scale.z = 0.05f;
+
+    DynaPolyActor_Init(&this->dyna, 0);
+    DynaPolyActor_LoadMesh(play, &this->dyna, &object_d_lift_Colheader_000F00);
+
+    this->dyna.actor.home.rot.y += 0x2000;
+    this->dyna.actor.home.rot.y &= 0xC000;
+    this->unk_29C = 20;
+
+    this->dyna.actor.shape.rot.y = this->dyna.actor.home.rot.y;
+    this->dyna.actor.world.rot.y = this->dyna.actor.home.rot.y;
+
+    for (val = 0, i = 0; i < this->unk_29C; i++, val += 7) {
+        temp = &this->unk_15C[i];
+
+        temp->unk_00.x = thisx->home.pos.x + (Math_SinS(thisx->shape.rot.y) * (i * 60));
+        temp->unk_00.y = thisx->home.pos.y;
+        temp->unk_00.z = thisx->home.pos.z + (Math_CosS(thisx->shape.rot.y) * (i * 60));
+        temp->unk_0C = val + 100;
+    }
+
+    func_809B1550(thisx, play);
+}
+
+void ObjChikuwa_Destroy(Actor* thisx, PlayState* play) {
+    ObjChikuwa* this = THIS;
+
+    DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+}
+
+void func_809B17D0(PlayState* play, ObjChikuwa* this, Vec3f* arg2) {
+    ObjChikuwaStruct2* temp_s0;
+    f32 var_fs2;
+    f32 temp;
+    s32 i;
+
+    for (i = 0, var_fs2 = -78.75f; i < 8; var_fs2 += 22.5f) {
+        temp_s0 = &this->unk_2A4[this->unk_9A4];
+
+        temp_s0->unk_00.x = (Rand_ZeroOne() * 0.05f) + 0.01f;
+        temp_s0->unk_00.y = (Rand_ZeroOne() * 0.16f) + 0.04f;
+        temp_s0->unk_00.z = (Rand_ZeroOne() * 0.05f) + 0.01f;
+
+        temp_s0->unk_0C.x = (Math_CosS(this->dyna.actor.shape.rot.y) * var_fs2) + arg2->x;
+        temp_s0->unk_0C.y = arg2->y;
+        temp_s0->unk_0C.z = (Math_SinS(this->dyna.actor.shape.rot.y) * var_fs2) + arg2->z;
+
+        temp_s0->unk_18 = (2.0f * Rand_ZeroOne()) - 1.0f;
+        temp_s0->unk_1C = (Rand_ZeroOne() * 4.0f) - 1.0f;
+        temp_s0->unk_20 = (2.0f * Rand_ZeroOne()) - 1.0f;
+
+        temp = Math3D_Vec3fMagnitude(&temp_s0->unk_00);
+        temp_s0->unk_24 = (-Rand_ZeroOne() * 0.5f) - temp * 25.0f;
+
+        temp_s0->unk_28.x = this->dyna.actor.shape.rot.x;
+        temp_s0->unk_28.y = this->dyna.actor.shape.rot.y;
+        temp_s0->unk_28.z = this->dyna.actor.shape.rot.z;
+
+        temp_s0->unk_2E.x = (Rand_ZeroOne() - 0.5f) * 5000.0f;
+        temp_s0->unk_2E.y = (Rand_ZeroOne() - 0.5f) * 5000.0f;
+        temp_s0->unk_2E.z = (Rand_ZeroOne() - 0.5f) * 5000.0f;
+
+        temp_s0->unk_34 = 80;
+        func_800B1210(play, &temp_s0->unk_0C, &D_809B1FDC, &D_809B1FD0, (Rand_ZeroOne() * 60.0f) + 80.0f,
+                      (Rand_ZeroOne() * 30.0f) + 60.0f);
+        this->unk_9A4 = (this->unk_9A4 + 1) & 0x1F;
+        i++;
```

Comment:
Just want to make sure, this can't go in the for loop directly?

Reply:
Nop, ends up flipping 2 instructions.